sched/addrenv/addrenv.c

```diff
@@ -400,9 +401,7 @@ int addrenv_restore(FAR struct addrenv_s *addrenv)
 
 void addrenv_take(FAR struct addrenv_s *addrenv)
 {
-  irqstate_t flags = enter_critical_section();
-  addrenv->refs++;
```

Comment:
Prototype of `refs` needs to be updated to `atomic_int`?

Reply:
The type of refs should not be the issue here as ordering should be guaranteed by atomic_fetch_add.